components/Tabs/Tabs.tsx

```diff
@@ -67,37 +139,101 @@ export const Tabs = ({ children }: TabsProps) => {
     [children]
   );
 
-  const labels = childTabs.map(({ props: { label } }) => label);
+  const dropdownVarsArr = useMemo(() => {
+    const dropdownVars: Set<string> = new Set();
+
+    childTabs.forEach(({ props: { inDropdown } }) => {
+      if (inDropdown) {
+        const dropdownFromItem = getDropdownFromItem(inDropdown);
+        dropdownFromItem.forEach((item) => dropdownVars.add(item));
+      }
+    });
+
+    return Array.from(dropdownVars).sort().concat(FAKE_DROPDOWN);
+  }, [childTabs]);
+
+  // making data which tabs to display in which dropdown options
+  // and which tabs should be selected in every dropdown
+  const tabsInDropdown: TabsInDropdowns = useMemo(() => {
+    const data: TabsInDropdowns = {};
+
+    for (const option of dropdownVarsArr) {
+      data[option] = [];
+      childTabs.forEach(({ props: { label, selected, inDropdown } }) => {
+        let dataTab: DataTab;
+        if (inDropdown && option !== FAKE_DROPDOWN) {
+          if (isInDropdown(inDropdown, option)) {
+            dataTab = { label, isPreSelected: Boolean(selected) };
+          }
+        } else {
+          dataTab = { label, isPreSelected: Boolean(selected) };
+        }
+        if (dataTab) {
+          data[option].push(dataTab);
+        }
+      });
+    }
+
+    return data;
+  }, [childTabs, dropdownVarsArr]);
 
-  const [currentLabel, setCurrentLabel] = useState(getSelectedLabel(childTabs));
+  const [selectedDropdownOption, setSelectedDropdownOpt] = useState(
+    dropdownSelected ? dropdownSelected : dropdownVarsArr[0]
+  );
+  const tabsMeta = tabsInDropdown[selectedDropdownOption];
+  const [currentTab, setCurrentTab] = useState(getSelectedTab(tabsMeta));
+
+  useEffect(() => {
+    setCurrentTab(getSelectedTab(tabsMeta));
+  }, [tabsMeta, selectedDropdownOption]);
 
   useEffect(() => {
     const scopedTab = childTabs.find(({ props }) =>
       getScopes(props.scope).includes(scope)
     );
 
     if (scopedTab) {
-      setCurrentLabel(scopedTab.props.label);
+      setCurrentTab(scopedTab.props.label);
     }
   }, [scope, childTabs]);
 
+  const visibleTabs = dropdownVarsArr.filter((t) => t !== FAKE_DROPDOWN);
+
   return (
     <div className={styles.wrapper}>
-      <div className={styles.header}>
-        {labels.map((label) => (
+      {visibleTabs.length ? (
```

Comment:
Instead of a ternary and `null`, try:

```ts
{visibleTabs.length && (<div className={styles["drop-wrapper"]}>
          <p className={styles["drop-title"]}>
            {dropdownCaption ? dropdownCaption : "Choose one of the options"}
          </p>
          <Dropdown
            className={styles.dropdown}
            value={selectedDropdownOption}
            options={visibleTabs}
            onChange={setSelectedDropdownOpt}
          />)
}

Reply:
Good catch! But but we need to append the `visibleTabs.length` conversion to a Boolean value, because if the array is empty, we will render 0 in HTML.

```
{Boolean(visibleTabs.length) && (
  ...
}
```